package.json

```diff
@@ -67,6 +67,7 @@
 		"@metamask/detect-provider": "^2.0.0",
 		"@walletconnect/web3wallet": "^1.14.0",
 		"alchemy-sdk": "^3.4.1",
+		"bottleneck": "^2.19.5",
```

Comment:
What's this?

Reply:
still a draft, but this [package](https://www.npmjs.com/package/bottleneck) "controls" how many requests per second or concurrent are done to a specific function; it will help to "bottleneck" requests to Etherescan and avoid touching the limit of calls/second

### Context

I am trying several solutions to solve the issue of the Etherescan limit calls/second...

Reply:
Let’s discuss Monday then. Adding any dependency introduces technical debt and potential security risks, especially when the library is not actively maintained, as is the case here. At the foundation—or at least within the Gix team—we generally avoid taking this route unless absolutely necessary. Additionally, I’d like to better understand the issue and explore the various solutions available.

Reply:
agreed!